### PR TITLE
Grow advanced UI columns to fill window

### LINKED ISSUE #22009: Advanced Web Interface - columns do not fill the width of the screen.
### Steps to reproduce the problem

1. Using firefox or chrome, log in to mastodon and enable the advanced web interface
2. Observe that columns do not fill the screen
3. Dive in to CSS and discover that the correct CSS exists, but is being overridden.

### Expected behaviour

Columns should fill the width of the screen

### Actual behaviour

Columns are a fixed 350px width

### Detailed description

See attached screenshot (content blurred out, developer mode enabled and highlighting the column elements):
Screenshot from 2022-12-03 11-02-25
(image description: a screenshot of the mastodon advanced web interface showing columns that do not fill the full width of the screen)

I found that if I uncheck (to disable) the "bad" CSS marked in the image below, then the "good" CSS is enabled, and the columns fill the width of the screen. This leads me to believe this is a bug. I tried to identify the exact line of code that is causing the issue so I could submit a PR, but I am finding that the line numbers listed in the developer window do not seem to me to correspond to the specific lines in question, so I had best not bang my head against the wall.

Screenshot from 2022-12-03 12-04-10
(image description: a zoomed in screenshot of the relevant CSS properties in the web inspector, with added markings showing the good CSS and the bad CSS that is overriding the good.)

### Specifications

Mastodon v4.0.2+trans_rights-f02cb1c-22326163749
Firefox 107
Ubuntu 22.04

### PR DESCRIPTION
**Pull request by pietvanzoen**
1 character change to have the advanced UI columns fill the available space. Tested this locally by making the same tweak in browser in Safari, Firefox, and Chrome.

This is my first ever PR on this project. Obviously I don't have all the context on how the components css interacts but this seems somewhat safe?

🙇 

## before

Screenshot 2022-11-09 at 10 06 26

## after
Screenshot 2022-11-09 at 10 07 15

shipit

### CODE REVIEW

**Comment by ineffyble** [comment]
Finally a use for my ultra-wide monitor, when I'm next at my desk. I would suspect it looks a little odd at UWHD ratio.

**Comment by melkcodes** [comment]
LGTM!

**Comment by pietvanzoen** [comment]
> Finally a use for my ultra-wide monitor, when I'm next at my desk. I would suspect it looks a little odd at UWHD ratio.

Haha. Perhaps. I don’t have access to a UW monitor. Feel free to try it out. Perhaps a max-width is needed?

**Comment by tbolon** [comment]
> > Finally a use for my ultra-wide monitor, when I'm next at my desk. I would suspect it looks a little odd at UWHD ratio.
> 
> Haha. Perhaps. I don’t have access to a UW monitor. Feel free to try it out. Perhaps a max-width is needed?

Yeah, in my instance I have simply changed the width to 450px (instead of 350px by default). Maybe a max-width could be appropriate?

**Comment by pietvanzoen** [comment]
> > > Finally a use for my ultra-wide monitor, when I'm next at my desk. I would suspect it looks a little odd at UWHD ratio.
> > 
> > 
> > Haha. Perhaps. I don’t have access to a UW monitor. Feel free to try it out. Perhaps a max-width is needed?
> 
> Yeah, in my instance I have simply changed the width to 450px (instead of 350px by default). Maybe a max-width could be appropriate?

makes sense. updated.

File: app/javascript/styles/mastodon/components.scss [hunk 1]
```diff
@@ -2414,6 +2415,7 @@ $ui-header-height: 55px;
 .columns-area--mobile {
   flex-direction: column;
   width: 100%;
+  max-width: 100%;
```
Comment (pietvanzoen) [inline app/javascript/styles/mastodon/components.scss:2418]: Wrong spot for this. Should be on `.column`

**Comment by PabloRubio21** [comment]
Looks good to me, but it should have a `max-width` and `min-width` just in case someone uses a very wide or very thin screen.

**Comment by mobilutz** [comment]
I have a similar idea, but a bit more flexible I think.
This is the pseudo CSS:
```css
.drawer, .column {
  flex: 1 0 auto;
  min-width: 300px;
}

.drawer {
  .max-width: 300px
}
```

Of course one could say to add a max-width to `.column` as well. I leave that up to the frontend people here.

Here some screenshots with the CSS from above.
Screenshot 2022-11-15 at 20 43 47
Screenshot 2022-11-15 at 20 44 07
Screenshot 2022-11-15 at 20 44 22

**Pull request closed without merging**